instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/TemporaryMetricsView.java

```diff
@@ -52,10 +44,72 @@
+      fullSet.add(SemanticAttributes.HTTP_URL);
+    } else if (containsAttribute(SemanticAttributes.HTTP_HOST, startAttributes, endAttributes)) {
+      // Use http.scheme, http.host and http.target
+      fullSet.add(SemanticAttributes.HTTP_SCHEME);
+      fullSet.add(SemanticAttributes.HTTP_HOST);
+      fullSet.add(SemanticAttributes.HTTP_TARGET);
+    } else if (containsAttribute(
+        SemanticAttributes.NET_PEER_NAME, startAttributes, endAttributes)) {
+      // Use http.scheme, net.peer.name, net.peer.port and http.target
+      fullSet.add(SemanticAttributes.HTTP_SCHEME);
+      fullSet.add(SemanticAttributes.NET_PEER_NAME);
+      fullSet.add(SemanticAttributes.NET_PEER_PORT);
+      fullSet.add(SemanticAttributes.HTTP_TARGET);
+    } else {
+      // Use http.scheme, net.peer.ip, net.peer.port and http.target
+      fullSet.add(SemanticAttributes.HTTP_SCHEME);
+      fullSet.add(SemanticAttributes.NET_PEER_IP);
+      fullSet.add(SemanticAttributes.NET_PEER_PORT);
+      fullSet.add(SemanticAttributes.HTTP_TARGET);
+    }
+    AttributesBuilder filtered = Attributes.builder();
+    applyView(filtered, startAttributes, fullSet);
+    applyView(filtered, endAttributes, fullSet);
+    return filtered.build();
+  }
+
+  private static <T> boolean containsAttribute(
+      AttributeKey<T> key, Attributes startAttributes, Attributes endAttributes) {
+    return startAttributes.get(key) != null || endAttributes.get(key) != null;
+  }
+
+  static Attributes applyServerDurationView(Attributes startAttributes, Attributes endAttributes) {
+    Set<AttributeKey> fullSet = new HashSet<>(durationAlwaysInclude);
+    // We pull identifying attributes according to:
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attribute-alternatives
+    if (containsAttribute(SemanticAttributes.HTTP_HOST, startAttributes, endAttributes)) {
+      // Use http.scheme, http.host and http.target
+      fullSet.add(SemanticAttributes.HTTP_SCHEME);
+      fullSet.add(SemanticAttributes.HTTP_HOST);
+      fullSet.add(SemanticAttributes.HTTP_TARGET);
+    } else if (containsAttribute(
+        SemanticAttributes.HTTP_SERVER_NAME, startAttributes, endAttributes)) {
+      // Use http.scheme, http.server_name, net.host.port, http.target
+      fullSet.add(SemanticAttributes.HTTP_SCHEME);
+      fullSet.add(SemanticAttributes.HTTP_SERVER_NAME);
+      fullSet.add(SemanticAttributes.HTTP_TARGET);
+    } else if (containsAttribute(
+        SemanticAttributes.NET_HOST_NAME, startAttributes, endAttributes)) {
+      // Use http.scheme, net.host.name, net.host.port, http.target
+      fullSet.add(SemanticAttributes.HTTP_SCHEME);
+      fullSet.add(SemanticAttributes.NET_HOST_NAME);
+      fullSet.add(SemanticAttributes.NET_HOST_PORT);
+      fullSet.add(SemanticAttributes.HTTP_TARGET);
+    } else {
+      // Use http.url
+      fullSet.add(SemanticAttributes.HTTP_URL);
```

Comment:
None of our HTTP server instrumentations use `http.url` - they always set the (scheme, host, target) triple

Reply:
Is that a guarantee or just the current implementation?

Reply:
Pretty much a guarantee if you're using the `...instrumenter.http` package - some time ago in #3700 (and related PRs) we've decided to simplify things and only set `http.url` in HTTP client instrumentations and (`http.scheme`, `http.host`, `http.target`) in server instrumentations.